horizon/scripts/onms-core-init.sh

```diff
@@ -601,3 +603,23 @@ else
  echo "We are unable to update Admin password. Exiting."
  exit 1
 fi
+
+if [ -d ${OVERLAY_CONFIG_MAPS} ]; then
+  echo "Processing overlay config maps ..."
+  # We need to make sure the directories are numerically sorted to match the configured configmap order.
+  for dir in $(ls -1d ${OVERLAY_CONFIG_MAPS}/* | sort -n); do
+    # When we first copy off of the configmap volume, we copy symlinks as files and ignore Kubernetes configmap volume ".." files.
+    # See: https://github.com/spring-projects/spring-boot/issues/23232
+    if [[ $(basename $dir) =~ .*-uncompress ]]; then
+      echo "  Copying files from $dir to ${OVERLAY_DIR}/ (and uncompressing .gz files) ..."
+      mkdir /tmp/uncompress
+      rsync -arO -L --exclude='..*' --no-perms --no-owner --no-group $dir/ /tmp/uncompress # don't bother outputting files copied yet
+      find /tmp/uncompress -name '*.gz' | xargs gzip -d -v 2>&1 | sed 's/^/    /'
```

Comment:
How about if we replace this with something like this

`      while IFS= read -r -d '' file
      do 
       tmpfolder=$(dirname "$file")
       echo "   Extracting $file to $tmpfolder"
       unzip -d "$tmpfolder" "$file" 2>&1
       rm "$file"
      done < <(find /tmp/uncompress -type f -name "*.zip" -print0)
`

Reply:
We paired and implemented ZIP support in a slightly different way.